### ISSUE TITLE
Crashes reported in HSBeaconPreviousMessagesController

### ISSUE THREAD

**Issue by jonklo** [helpscout/beacon-ios-sdk#33]
* Beacon install method: [X] Carthage, [ ] CocoaPods, [ ] Manual
* Version: 1.0.1

Our crash reporting tool has reported two issues with `HSBeaconPreviousMessagesController` in the SDK. Including the tracebacks here:

### HSBeaconPreviousMessagesController prepareForSegue:sender

```
Fatal Exception: NSRangeException
-[__NSArray0 objectAtIndex:]: index 0 beyond bounds for empty NSArray

0  CoreFoundation                 0x1c38e527c __exceptionPreprocess
1  libobjc.A.dylib                0x1c2abf9f8 objc_exception_throw
2  CoreFoundation                 0x1c37fa7bc -[__NSCFArray count]
3  Beacon                         0x100a5e4c0 -[HSBeaconPreviousMessagesController prepareForSegue:sender:] + 196 (HSBeaconPreviousMessagesController.m:196)
4  UIKitCore                      0x1f0378914 -[UIStoryboardSegueTemplate _performWithDestinationViewController:sender:]
5  UIKitCore                      0x1f0378824 -[UIStoryboardSegueTemplate _perform:]
6  UIKitCore                      0x1efc44ab4 -[UIViewController performSegueWithIdentifier:sender:]
7  Beacon                         0x100a5e28c -[HSBeaconPreviousMessagesController tableView:didSelectRowAtIndexPath:] + 169 (HSBeaconPreviousMessagesController.m:169)
8  UIKitCore                      0x1f03ee060 -[UITableView _selectRowAtIndexPath:animated:scrollPosition:notifyDelegate:]
9  UIKitCore                      0x1f03ee2c4 -[UITableView _userSelectRowAtPendingSelectionIndexPath:]
10 UIKitCore                      0x1f01f6dec _runAfterCACommitDeferredBlocks
11 UIKitCore                      0x1f01e593c _cleanUpAfterCAFlushAndRunDeferredBlocks
12 UIKitCore                      0x1f02127ac _afterCACommitHandler
13 CoreFoundation                 0x1c38765f8 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__
14 CoreFoundation                 0x1c3871320 __CFRunLoopDoObservers
15 CoreFoundation                 0x1c387189c __CFRunLoopRun
16 CoreFoundation                 0x1c38710b0 CFRunLoopRunSpecific
17 GraphicsServices               0x1c5a7179c GSEventRunModal
18 UIKitCore                      0x1f01eb978 UIApplicationMain
20 libdyld.dylib                  0x1c33368e0 start
```

### HSBeaconPreviousMessagesController tableView:cellForRowAtIndexPath:
```
Fatal Exception: NSRangeException
 -[__NSArray0 objectAtIndex:]: index 3 beyond bounds for empty NSArray
0  CoreFoundation                 0x22343d27c __exceptionPreprocess
1  libobjc.A.dylib                0x2226179f8 objc_exception_throw
2  CoreFoundation                 0x2233527bc -[__NSCFArray count]
3  Beacon                         0x10548209c -[HSBeaconPreviousMessagesController tableView:cellForRowAtIndexPath:] + 148 (HSBeaconPreviousMessagesController.m:148)
4  UIKitCore                      0x24ff62358 -[UITableView _createPreparedCellForGlobalRow:withIndexPath:willDisplay:]
5  UIKitCore                      0x24ff62858 -[UITableView _createPreparedCellForGlobalRow:willDisplay:]
6  UIKitCore                      0x24ff2f060 -[UITableView _updateVisibleCellsNow:isRecursive:]
7  UIKitCore                      0x24ff4c380 -[UITableView layoutSubviews]
8  UIKitCore                      0x2501daed0 -[UIView(CALayerDelegate) layoutSublayersOfLayer:]
9  QuartzCore                     0x2278f1a20 -[CALayer layoutSublayers]
10 QuartzCore                     0x2278f69c8 CA::Layer::layout_if_needed(CA::Transaction*)
11 QuartzCore                     0x2278592d0 CA::Context::commit_transaction(CA::Transaction*)
12 QuartzCore                     0x227887330 CA::Transaction::commit()
13 QuartzCore                     0x227887f20 CA::Transaction::observer_callback(__CFRunLoopObserver*, unsigned long, void*)
14 CoreFoundation                 0x2233ce5f8 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__
15 CoreFoundation                 0x2233c9320 __CFRunLoopDoObservers
16 CoreFoundation                 0x2233c989c __CFRunLoopRun
17 CoreFoundation                 0x2233c90b0 CFRunLoopRunSpecific
18 GraphicsServices               0x2255c979c GSEventRunModal
19 UIKitCore                      0x24fd43978 UIApplicationMain
21 libdyld.dylib                  0x222e8e8e0 start
```

**Comment by samsymons** [comment]
@jonklo Thanks for the crash report – we'll investigate this and close this issue once this is fixed!

**Comment by samsymons** [comment]
Hey @jonklo! Just quickly following up before we begin investigating this, do you happen to have reproduction steps for these crashes?

**Comment by jonklo** [comment]
Managed to get it to crash repeatedly now:
1. Navigate to the "Previous conversations" screen
2. Open a message
3. Close the message, and start scrolling immediately after

Hope that helps you track down the bug, best of luck!

**Comment by ScottPetit** [comment]
Hey @jonklo,

Thanks so much for the great reproduction steps.  I wanted to let you know that this should be resolved in our latest SDK release version 1.0.2.
